src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -524,6 +556,95 @@ mongoc_client_encryption_encrypt_opts_set_query_type (
    opts->query_type = query_type ? bson_strdup (query_type) : NULL;
 }
 
+/*--------------------------------------------------------------------------
+ * Explicit Encryption Range Options
+ *--------------------------------------------------------------------------
+ */
+mongoc_client_encryption_encrypt_range_opts_t *
+mongoc_client_encryption_encrypt_range_opts_new (void)
+{
+   return bson_malloc0 (sizeof (mongoc_client_encryption_encrypt_range_opts_t));
+}
+
+void
+mongoc_client_encryption_encrypt_range_opts_set_sparsity (
+   mongoc_client_encryption_encrypt_range_opts_t *range_opts, int64_t sparsity)
+{
+   if (!range_opts) {
+      return;
+   }
+   range_opts->sparsity = sparsity;
+}
+
+void
+mongoc_client_encryption_encrypt_range_opts_set_min_max (
+   mongoc_client_encryption_encrypt_range_opts_t *range_opts,
+   bson_value_t min,
+   bson_value_t max)
+{
+   BSON_ASSERT_PARAM (&min);
+   BSON_ASSERT_PARAM (&max);
+
+   if (!range_opts) {
+      return;
+   }
+
+   range_opts->min.set = true;
+   range_opts->min.value = min;
+   range_opts->max = max;
+}
+
+void
+mongoc_client_encryption_encrypt_range_opts_set_precision (
+   mongoc_client_encryption_encrypt_range_opts_t *range_opts, int32_t precision)
+{
+   BSON_ASSERT_PARAM (&precision);
+
+   if (!range_opts) {
+      return;
+   }
+   range_opts->precision.set = true;
+   range_opts->precision.value = precision;
+}
+
+void
+mongoc_client_encryption_encrypt_opts_set_range_opts (
+   mongoc_client_encryption_encrypt_opts_t *opts,
+   mongoc_client_encryption_encrypt_range_opts_t *range_opts)
+{
+   if (!opts) {
+      return;
+   }
+   if (range_opts) {
+      opts->range_opts.set = true;
+      opts->range_opts.value = range_opts;
+   }
+}
+
+void
```

Comment:
```suggestion
static void
```

Prefer `static` on functions that are not called outside of the file. `static` prevents other files from linking to the function.